crates/sui-sdk/examples/utils.rs

```diff
@@ -0,0 +1,305 @@
+    // that we will submit to the network
+    let tx_data = TransactionData::new_programmable(
+        *sender,
+        vec![coin.object_ref()],
+        builder,
+        max_gas_budget,
+        gas_price,
+    );
+
+    // sign & execute the transaction
+    let keystore = FileBasedKeystore::new(&sui_config_dir()?.join(SUI_KEYSTORE_FILENAME))?;
+    let signature = keystore.sign_secure(sender, &tx_data, Intent::sui_transaction())?;
+
+    let transaction_response = sui
+        .quorum_driver_api()
+        .execute_transaction_block(
+            Transaction::from_data(tx_data, Intent::sui_transaction(), vec![signature]),
+            SuiTransactionBlockResponseOptions::new(),
+            Some(ExecuteTransactionRequestType::WaitForLocalExecution),
+        )
+        .await?;
+    Ok(transaction_response.digest)
+}
+
+pub async fn retrieve_wallet() -> Result<WalletContext, anyhow::Error> {
+    let wallet_conf = sui_config_dir()?.join(SUI_CLIENT_CONFIG);
+    let keystore_path = sui_config_dir()?.join(SUI_KEYSTORE_FILENAME);
+
+    // check if a wallet exists and if not, create a wallet and a sui client config
+
+    if !keystore_path.exists() {
+        let keystore = FileBasedKeystore::new(&keystore_path)?;
+        keystore.save()?;
+    }
+
+    if !wallet_conf.exists() {
+        let keystore = FileBasedKeystore::new(&keystore_path)?;
+        let mut client_config = SuiClientConfig::new(keystore.into());
+        client_config.add_env(SuiEnv {
+            alias: "testnet".to_string(),
+            rpc: "https://fullnode.testnet.sui.io:443".to_string(),
+            ws: None,
+        });
+
+        client_config.add_env(SuiEnv::devnet());
+
+        client_config.add_env(SuiEnv {
+            alias: "localnet".to_string(),
+            rpc: "http://127.0.0.1:9000".to_string(),
+            ws: None,
+        });
+
+        if client_config.active_env.is_none() {
+            client_config.active_env = client_config.envs.first().map(|env| env.alias.clone());
+        }
+
+        client_config.save(&wallet_conf)?;
+        info!("Client config file is stored in {:?}.", &wallet_conf);
+    }
+
+    let mut keystore = FileBasedKeystore::new(&keystore_path)?;
+    let mut client_config: SuiClientConfig = PersistedConfig::read(&wallet_conf)?;
+
+    if keystore.addresses().is_empty() {
+        keystore.generate_and_add_new_key(ED25519, None, None)?;
+        keystore.generate_and_add_new_key(ED25519, None, None)?;
+    } else if keystore.addresses().len() < 2 {
+        keystore.generate_and_add_new_key(ED25519, None, None)?;
+    }
+
+    if client_config.active_address.is_none() {
+        let addresses = keystore.addresses();
+        let active_address = addresses
+            .get(0)
+            .expect("Should have at least two Sui addresses, but cannot find one. Aborting");
+        client_config.active_address = Some(*active_address);
+        client_config.save(&wallet_conf)?;
+    }
```

Comment:
nit: Can neaten up the control-flow here


```suggestion
    let default_active_address = if let Some(address) = keystore.addresses().first() {
        address
    } else {
        keystore.generate_and_add_new_key(ED25519, None, None)?.0
    };
    
    if keystore.addresses().len() < 2 {
        keystore.generate_and_add_new_key(ED25519, None, None)?;
    }

    if client_config.active_address.is_none() {
        client_config.active_address = Some(default_active_address);
        client_config.save(&wallet_conf)?;
    }
```